includes/core/DataTransferObjects/API/class-response.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Response DTO.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\DataTransferObjects;
+
+use WCPay\Core\Contracts\API\Response\Base_Response;
+use WP_Http_Cookie;
+
+/**
+ * DTO for parsing response from WCPay server API.
+ */
+final class Response extends Data_Transfer_Object implements Base_Response {
+	/**
+	 * Response body.
+	 *
+	 * @var string $body
+	 */
+	private $body;
+
+	/**
+	 * Status code.
+	 *
+	 * @var int $code
+	 */
+	private $code;
+
+	/**
+	 * Response headers.
+	 *
+	 * @var array $headers
+	 */
+	private $headers;
+
+	/**
+	 * Response message.
+	 *
+	 * @var string $message
+	 */
+	private $message;
+
+	/**
+	 * Response cookies.
+	 *
+	 * @var WP_Http_Cookie[] $cookies
+	 */
+	private $cookies;
+
+	/**
+	 * Parsed response array.
+	 *
+	 * @var array $response_dody
+	 */
+	private $response_data;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param string           $body Response body.
+	 * @param int              $code Response code.
+	 * @param array            $headers Response headers.
+	 * @param string           $message Response message.
+	 * @param WP_Http_Cookie[] $cookies Response cookies.
+	 */
+	public function __construct( $body, $code, $headers, $message, $cookies ) {
+		$this->body          = $body;
+		$this->code          = $code;
+		$this->headers       = $headers;
+		$this->message       = $message;
+		$this->cookies       = $cookies;
+		$this->response_data = json_decode( $body, true );
+	}
+
+	/**
+	 * Returns true if response is JSON.
+	 *
+	 * @return bool
+	 */
+	public function is_json_response() {
+		return 'application/json' === substr( $this->headers['content-type'] ?? '', 0, strlen( 'application/json' ) );
```

Comment:
Isn't this the same as:

```suggestion
		return 0 === strpos( $this->headers['content-type'] ?? '', 'application/json' );
```